### PR TITLE
parse help message, new test search based on the fields

### PR DESCRIPTION
**Pull request by tstrych**
I think with a bit of work test like this can be added to all entities. 
I would like to move parse method somewhere else but not sure where :neutral_face: 

Also firstly I made the test the different way. I asserted all the fields with all hosts but, that can be potentially really time consuming if sat has a lot of hosts. So I decided to add steps 1 and 4 and use only one host.

The `parse_two_columns` method is not necessary, but I though maybe it can be useful in future if we want to check also options from help messages. Let me know what you think about the way I parse this. 

Test result:
```
tests/foreman/cli/test_host.py::test_positive_search_all_field_sets
=========== 1 passed, 55 deselected, 4 warnings in 76.35s (0:01:16) ============
```

### CODE REVIEW

**Comment by JacobCallahan** [comment]
I think we should have a larger team conversation about whether or not we want to test this kind of behavior in robottelo.

**Comment by tstrych** [comment]
Two possibilities, merge this as it is already written or close the pull request.

**Comment by JacobCallahan** [comment]
@tstrych did you get a response from dev about their coverage of this?

**Comment by tstrych** [comment]
upstream PR with the test, where you just check if the category name starts with capital letter
https://github.com/theforeman/hammer-cli/pull/342/files#diff-ee9ab7035ec10719ed2f749114a402e46d52f3c1f5cff9f5b10f3358b71abcfe

help test: if the help message looks same 
https://github.com/theforeman/hammer-cli/blob/master/test/functional/help_test.rb

if hammer host list shows column
https://github.com/theforeman/hammer-cli-foreman/blob/master/test/unit/host_test.rb#L26

there is no test which tests the functionality like this one, search all fields dynamically, 
hammer help do not use API call, so it can run without internet connection, so the unit tests have no data for 

I am thinking if this test is bringing more value than mock. 
And how much added value it have, compared to work with some changes. 
Right now:  feeling yes, let's try that. Give it a chance. 
But later on: please remove it please just because I have other things than just fixing the help test.

**Review by pondrejk — APPROVED** [review]

**Comment by jyejare** [comment]
@tstrych I agree with @JacobCallahan that it needs a larger discussion on having the `help` strings tests in robottelo. TBH, Robottelo is a pure functional testing framework, we don't worry much about the usability testing here in this framework just as this PR and testing the `help` text.

There is also a localization and internationalization team, might be testing these help texts. But altogether I wont invest in testing the help text at all at least in robottelo.

**Comment by tstrych** [comment]
Understand. Definitely I think, it's not good idea to spread this approach across other entities. But it's already been implemented, why not to give it a chance. And to experiment a bit with this approach. 

But I understand if you don't want to. The reasoning you mentioned makes sense to me. I have no problem if this PR will be closed. Somebody should determine that. I can ask on next joint meeting.

File: tests/foreman/cli/test_host.py [hunk 1]
```diff
@@ -152,6 +152,161 @@ def function_user(function_host):
...
+        try:
+            if options_start_with_dash and line_lst[0].startswith('-'):
+                options[line_lst[0]] = line_lst[1]
+                previous_line = line_lst[0]
+            else:
+                options[line_lst[0]] = line_lst[1]
+        except IndexError:
+            if options_start_with_dash and previous_line is not None:
+                # append message to previous line value
+                options[previous_line] = options[previous_line] + ' ' + line_lst[0]
```
Comment (mshriver) [inline tests/foreman/cli/test_host.py:202]: I think f-string would work well here instead of concatenating.

**Review by mshriver — APPROVED** [review]
I'm leaving this one up to @JacobCallahan to merge or not, but I'm up for letting @tstrych utilize this test in 6.10 and determine whether its providing value.

**Comment by JacobCallahan** [comment]
We'll merge this one and see how it goes. For the moment this is neither an encouraged or discouraged practice.

**Review by JacobCallahan — APPROVED** [review]

**Pull request merged**